purchase_operating_unit/models/purchase_order.py

```diff
@@ -14,14 +14,15 @@ def _default_picking_type(self):
         res = super(PurchaseOrder, self)._default_picking_type()
         type_obj = self.env["stock.picking.type"]
         operating_unit = self.env["res.users"].operating_unit_default_get(self.env.uid)
-        types = type_obj.search(
-            [
-                ("code", "=", "incoming"),
-                ("warehouse_id.operating_unit_id", "=", operating_unit.id),
-            ]
-        )
-        if types:
-            res = types[:1].id
+        if operating_unit:
+            types = type_obj.search(
+                [
+                    ("code", "=", "incoming"),
+                    ("warehouse_id.operating_unit_id", "=", operating_unit.id),
+                ]
+            )
+            if types:
+                res = types[0].id
```

Comment:
Seemed cleaner to me, but I can revert it.

`[:1]` is useful to make sure there are no errors on an empty recordset. But we already have `if types`, and also it would error out when trying to access `.id` anyway.

Either way, that's not the point of the issue, so I'll remove it.